Mathlib/CategoryTheory/Limits/Constructions/Filtered.lean

```diff
@@ -32,10 +32,12 @@ namespace CategoryTheory.Limits
 
 namespace CoproductsFromFiniteFiltered
 
+variable [HasFiniteCoproducts C]
+
 /-- If `C` has finite coproducts, a functor `Discrete α ⥤ C` lifts to a functor
     `Finset (Discrete α) ⥤ C` by taking coproducts. -/
 @[simps!]
-def liftToFinset [HasFiniteCoproducts C] (F : Discrete α ⥤ C) : Finset (Discrete α) ⥤ C where
+def liftToFinsetObj (F : Discrete α ⥤ C) : Finset (Discrete α) ⥤ C where
```

Comment:
This is perhaps orthogonal to this PR, but it seems to me that we should instead have `liftToMultiset` and compose with the forgetful functor Multiset => Finset where required. The constructions here should all go through for multisets...

Reply:
Hmm.. multisets aren't used at all in the category theory library so far, which might or not be because we want swaps of two elements in the diagram to induce an isomorphism and not an equality between the (co)products.

Taking `Fin n` as the  the diagram category does have that effect via `CategoryTheory.Limits.Sigma.reindex`, but taking `Multiset` would destroy it.

Reply:
I'm not quite sure I follow. Finsets are just multisets without duplicates (in a literal sense), they should have identical behaviour when it comes to swaps. 

The difference arises if you take a product over something like {X, X}, in which the finset version would give (something isomorphic to) X, whereas the multiset version would give X * X.

Reply:
(Note that `sigmaObj` takes an arbitary family, which, if indexed over `Nat` is also behaving like a `List` and not `Multiset`)

Reply:
Right, this is why I think `Multiset` is a more sensible choice! It lives exactly between `List` and `Finset`, and the observation that this construction is going from List to Finset is an argument that it could be generalised to instead go from List to Multiset, and later to Finset

Reply:
Got it, I think you're right. Might be a good refinement for a later PR...

Reply:
Yes, admittedly this PR was the wrong place to bring up this discussion!